### ISSUE TITLE
[question]Can not start container other than the one containing redis.

### ISSUE THREAD

**Issue by napolenli** [hashicorp/nomad#2865]
Hi I have been trying to do some experiments with nomad for my company, since we encountered tons of problem with swarm, especially network issues.

Everything is fine when I follow the instructions from nomad Get Started. However, when I was trying to run a different container other than example.nomad, such as mysql.nomad, I could not start it, which makes it even harder to run our own app with tons of args (tried args in config, didn't work neither.), of course, let's resolve the simple problem first.

### Nomad version
Nomad v0.5.6

### Operating system and Environment details
CentOS Linux release 7.3.1611 (Core)

### Issue
Could not start mysql container in this version of nomad.

### Reproduction steps
nomad run mysql.nomad, please change the file name from mysql.nomad.txt to mysql.nomad, since github does not support file extension as .nomad.

### Nomad Server logs (if appropriate)
Not even logs produced while trying to start mysql container with nomad.

### Job file (if appropriate)
[mysql.nomad.txt](https://github.com/hashicorp/nomad/files/1160099/mysql.nomad.txt)

**Comment by shantanugadgil** [comment]
Hi @napolenli ,
I would say "don't get intimated by the job file, it only feels complicated at the beginning. 😀 

That said, what output do you get after executing "nomad run mysql.nomad" ?

If you see an allocation id, you can use that to see what happened to the allocation.

* nomad alloc-status <allocation-id>
* nomad logs <allocation-id>

**Comment by napolenli** [comment]
Hi @shantanugadgil 
Thank you for responding so fast, here are the answers to your questions.
## 1). Output after executing nomad run mysql.nomad.
==> Monitoring evaluation "2d121227"
    Evaluation triggered by job "mysql"
    Allocation "5e9a49b0" created: node "988e99dd", group "cache"
    Evaluation status changed: "pending" -> "complete"
==> Evaluation "2d121227" finished with status "complete"
## 2). nomad alloc-status 5e9a49b0
ID                  = 5e9a49b0
Eval ID             = 2d121227
Name                = mysql.cache[0]
Node ID             = 988e99dd
Job ID              = mysql
Client Status       = pending
Client Description  = <none>
Desired Status      = run
Desired Description = <none>
Created At          = 07/20/17 04:10:32 CST

Task "mysql" is "pending"
Task Resources
CPU      Memory   Disk     IOPS  Addresses
500 MHz  256 MiB  300 MiB  0     db: 10.0.50.57:43685

Recent Events:
Time                   Type        Description
07/20/17 04:11:29 CST  Restarting  Task restarting in 25.766924194s
07/20/17 04:11:29 CST  Terminated  Exit Code: 1, Exit Message: "Docker container exited with non-zero exit code: 1"
07/20/17 04:11:28 CST  Started     Task started by client
07/20/17 04:10:32 CST  Driver      Downloading image mysql:8.0.1
07/20/17 04:10:32 CST  Task Setup  Building Task Directory
07/20/17 04:10:32 CST  Received    Task received by client
## 3). nomad logs 
Nothing was in the output after executing this command.

Thanks!

**Comment by shantanugadgil** [comment]
All seems fine upto the point where Nomad started the Docker.
What seems to be failing is the MySQL start.

What I would recommend is to `enable_syslog` on the Nomad Agent (10.0.50.57 in your case) and check what appears in `/var/log/messages` (assuming the agent is CentOS as well)

I am familiar with starting MySQL as a service, though I have never started a MySQL Docker.
What I think is missing is the startup parameters (or environment flags) as the following seems to indicate:
https://hub.docker.com/_/mysql/

Regards,
Shantanu

**Comment by dadgar** [comment]
@shantanugadgil Appreciate you jumping in to help 👍 
@napolenli I ran the job and saw the same exit code 1. The next thing I did was run `nomad logs <alloc-id>` to see the output of the run and this is what is outputted.

```
nomad logs cf24ebf1
error: database is uninitialized and password option is not specified
  You need to specify one of MYSQL_ROOT_PASSWORD, MYSQL_ALLOW_EMPTY_PASSWORD and MYSQL_RANDOM_ROOT_PASSWORD
```

So looks like you need to set some environment variables. You can do that via: https://www.nomadproject.io/docs/job-specification/env.html

I am going to close this but if you have more questions feel free to continue to post here. In the future you may consider Gitter or the mailing list.

**Comment by napolenli** [comment]
Hi @shantanugadgil  thank you so much for helping me out.
@dadgar I tried your advice, it worked! Thank you so much for the help!
But I have one more question, @dadgar , how did you get the log output like that, when I was trying to run nomad logs <alloc-id>, I got the following

> Error reading file: Unexpected response code: 404 (task "mysql" not started yet. No logs available)

What kind of tweak do I need to do to get the log as what you got ?

Thanks!
William
